src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -125,6 +137,29 @@ public void Stop(TimeSpan timeout)
             }
         }
 
+        private void DisposeConnections(TimeSpan timeout)
+        {
+            try
+            {
+                // Close and wait for all connections
+                if (!ConnectionManager.WalkConnectionsAndClose(timeout))
+                {
+                    _log.LogError(0, null, "Waiting for connections timed out");
+                }
+
+                PostAsync(state =>
+                {
+                    var listener = (KestrelThread)state;
+                    listener.WriteReqPool.Dispose();
+                },
+                this).Wait();
```

Comment:
This will cause `KestrelThread.Stop` to hang if the thread is unresponsive. This should never happen if the thread is still alive, but it's still possible if something else is blocking the event loop.

It might be OK to block the caller to `Stop()` in this case, but it's a change in behavior. Before, only an error would be logged.